### ISSUE TITLE
2*A100，lora，sft Step-Audio-2-mini，OOM

### ISSUE THREAD

**Issue by wulaoshi** [modelscope/ms-swift#5896]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
transformers==4.55.4
ms_swift==3.8.1

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

你好，我在使用2*A100 lora训练 Step-Audio-2-mini 时爆显存。
以下是训练参数配置：
` 
nproc_per_node=2

CUDA_VISIBLE_DEVICES=6,7 \
ENABLE_AUDIO_OUTPUT=0 \
NPROC_PER_NODE=$nproc_per_node \
VIDEO_MAX_PIXELS=50176 \
FPS_MAX_FRAMES=12 \
MAX_PIXELS=1003520 \
/home/minicondas/envs/zy/bin/swift sft \
    --model /data/models/stepfun-ai/Step-Audio-2-mini \
    --dataset ourdata.jsonl \
    --split_dataset_ratio 0 \
    --train_type lora \
    --torch_dtype bfloat16 \
    --num_train_epochs 3 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --learning_rate 3.0e-5 \
    --strict False \
    --lora_rank 16 \
    --lora_alpha 32 \
    --freeze_vit true \
    --freeze_aligner true \
    --gradient_accumulation_steps $(expr 64 / $nproc_per_node) \
    --eval_steps 50 \
    --save_steps 134 \
    --save_total_limit 2 \
    --logging_steps 5 \
    --max_length 8192 \
    --output_dir test \
    --warmup_ratio 0 \
    --truncation_strategy delete\
    --dataloader_num_workers 16 \
    --deepspeed zero3
`

日志里打印的实际训练参数量：model_parameter_info: PeftModelForCausalLM: 8355.5494M Params (40.3702M Trainable [0.4832%]), 0.0001M Buffers.

按理说在此情况下，8B的模型不应该爆才对，怀疑有bug。

**Comment by Jintao-Huang** [comment]
Try using the main branch — this model isn't supported in 3.8.1 yet.

**Comment by wulaoshi** [comment]
> Try using the main branch — this model isn't supported in 3.8.1 yet.

Yes, I forgot to mention—I actually pulled the latest updates from the main branch via the web interface and ran it. After debugging, I confirmed that ms_swift already supports Step-Audio-2-mini and includes relevant code like:
```
register_template(
    TemplateMeta(
        MLLMTemplateType.step_audio2_mini,
        template_cls=StepAudio2MiniTemplate,
    ....
```

I modified it as follows:

```
register_template(
    QwenTemplateMeta(
        MLLMTemplateType.step_audio2_mini,
        template_cls=StepAudio2MiniTemplate,
        prefix=[],
        prompt=['<|BOT|>human\n{{QUERY}}<|EOT|><|BOT|>assistant\n'],
        system_prefix=['<|BOT|>system\n{{SYSTEM}}<|EOT|>'],
        chat_sep=['<|EOT|>'],
        suffix=['<|EOT|>'],
    )
)
```

After that, I also checked the `input_ids`, the corresponding loss, and their lengths, and didn’t find any obvious problems. I’m not sure what’s causing the OOM issue.

**Comment by Qoboty** [comment]
> **Describe the bug** What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
> 
> **Your hardware and system info** Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等) transformers==4.55.4 ms_swift==3.8.1
> 
> **Additional context** Add any other context about the problem here(在这里补充其他信息)
> 
> 你好，我在使用2*A100 lora训练 Step-Audio-2-mini 时爆显存。 以下是训练参数配置： ` nproc_per_node=2
> 
> CUDA_VISIBLE_DEVICES=6,7 ENABLE_AUDIO_OUTPUT=0 NPROC_PER_NODE=$nproc_per_node VIDEO_MAX_PIXELS=50176 FPS_MAX_FRAMES=12 MAX_PIXELS=1003520 /home/minicondas/envs/zy/bin/swift sft --model /data/models/stepfun-ai/Step-Audio-2-mini --dataset ourdata.jsonl --split_dataset_ratio 0 --train_type lora --torch_dtype bfloat16 --num_train_epochs 3 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --learning_rate 3.0e-5 --strict False --lora_rank 16 --lora_alpha 32 --freeze_vit true --freeze_aligner true --gradient_accumulation_steps $(expr 64 / $nproc_per_node) --eval_steps 50 --save_steps 134 --save_total_limit 2 --logging_steps 5 --max_length 8192 --output_dir test --warmup_ratio 0 --truncation_strategy delete --dataloader_num_workers 16 --deepspeed zero3 `
> 
> 日志里打印的实际训练参数量：model_parameter_info: PeftModelForCausalLM: 8355.5494M Params (40.3702M Trainable [0.4832%]), 0.0001M Buffers.
> 
> 按理说在此情况下，8B的模型不应该爆才对，怀疑有bug。

@wulaoshi  Can you show some line in your ourdata.jsonl file, 'head -n 5 ourdata.jsonl'

**Comment by slin000111** [comment]
> **Describe the bug** What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
> 
> **Your hardware and system info** Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等) transformers==4.55.4 ms_swift==3.8.1
> 
> **Additional context** Add any other context about the problem here(在这里补充其他信息)
> 
> 你好，我在使用2*A100 lora训练 Step-Audio-2-mini 时爆显存。 以下是训练参数配置： ` nproc_per_node=2
> 
> CUDA_VISIBLE_DEVICES=6,7 ENABLE_AUDIO_OUTPUT=0 NPROC_PER_NODE=$nproc_per_node VIDEO_MAX_PIXELS=50176 FPS_MAX_FRAMES=12 MAX_PIXELS=1003520 /home/minicondas/envs/zy/bin/swift sft --model /data/models/stepfun-ai/Step-Audio-2-mini --dataset ourdata.jsonl --split_dataset_ratio 0 --train_type lora --torch_dtype bfloat16 --num_train_epochs 3 --per_device_train_batch_size 1 --per_device_eval_batch_size 1 --learning_rate 3.0e-5 --strict False --lora_rank 16 --lora_alpha 32 --freeze_vit true --freeze_aligner true --gradient_accumulation_steps $(expr 64 / $nproc_per_node) --eval_steps 50 --save_steps 134 --save_total_limit 2 --logging_steps 5 --max_length 8192 --output_dir test --warmup_ratio 0 --truncation_strategy delete --dataloader_num_workers 16 --deepspeed zero3 `
> 
> 日志里打印的实际训练参数量：model_parameter_info: PeftModelForCausalLM: 8355.5494M Params (40.3702M Trainable [0.4832%]), 0.0001M Buffers.
> 
> 按理说在此情况下，8B的模型不应该爆才对，怀疑有bug。

去掉deepspeed zero3试试
